demos/sjtwo/ltc4150/source/main.cpp

```diff
@@ -0,0 +1,53 @@
+#include "L1_Peripheral/lpc40xx/gpio.hpp"
+#include "L2_HAL/sensors/battery/ltc4150.hpp"
+#include "utility/log.hpp"
+#include "utility/time.hpp"
+#include "utility/units.hpp"
+
+bool ApproxEquality(float expected, float actual, float resolution)
+{
+  return -resolution < (actual - expected) && (actual - expected) < resolution;
+}
+
+int main()
+{
+  constexpr float kResolution = 0.001f;
+  LOG_INFO("Ltc4150 application starting...");
+  // Creating GPIO on pin 2.0
+  sjsu::lpc40xx::Gpio tick_pin(2, 0);
+  // Creating GPIO on pin 2.1
+  sjsu::lpc40xx::Gpio polarity_pin(2, 1);
+  // Set the sense resistor resistance to be 50 milliohms. This will provide a
+  // 50 millivolt drop at maximum charge or discharge to invoke an interrupt on
+  // the tick_pin.
+  units::impedance::ohm_t resistance = 0.00050_Ohm;
+  // Initialize the LTC4150 with the two GPIO pins and resistance.
+  sjsu::Ltc4150 counter(tick_pin, polarity_pin, resistance);
+  // Initialize the LTC4150 class, setting two GPIO pins to be outputs and
+  // attaching interrupts.
+  counter.Initialize();
+  // Set the previous charge to be 0 mAh.
+  units::charge::milliampere_hour_t previous_charge = 0_mAh;
+
+  while (true)
+  {
+    // Grab the calculated charge from the counter, based on how many ticks it
+    // received and the polarity.
+    units::charge::milliampere_hour_t current_charge = counter.GetCharge();
```

Comment:
```suggestion
    units::charge::milliampere_hour_t recent_charge = counter.GetCharge();
```
The name "current" here is kinda confusing as we are dealing with electrical current and temporal current.